share/chruby/chruby.sh

```diff
@@ -46,13 +46,14 @@ function chruby_use()
 	eval "$(RUBYGEMS_GEMDEPS="" "$RUBY_ROOT/bin/ruby" - <<EOF
 puts "export RUBY_ENGINE=#{Object.const_defined?(:RUBY_ENGINE) ? RUBY_ENGINE : 'ruby'};"
 puts "export RUBY_VERSION=#{RUBY_VERSION};"
+puts "export RUBY_ENGINE_VERSION=#{defined?(RUBY_ENGINE_VERSION) ? RUBY_ENGINE_VERSION : RUBY_VERSION};"
```

Comment:
I think `defined?(RUBY_ENGINE_VERSION)` here should be `Object.const_defined?(:RUBY_ENGINE_VERSION)`, since mruby doesn't implement the `defined?` keyword.

Reply:
Good point, fixed.
I wonder if we even need to check that `RUBY_ENGINE_VERSION` is defined, it's been there for a long time, but probably better safe than sorry.

I tried with MRuby 1.4.0 and 2.0.0, and it works except `chruby mruby` shows:
```
trace (most recent call last):
	[0] -:4
-:4: uninitialized constant LoadError (NameError)
```
(which is the same on `master`)

Reply:
I was glad you checked, since `RUBY_ENGINE_VERSION` was added to CRuby in 2.3.0.